checkov/terraform/graph_builder/variable_rendering/evaluate_terraform.py

```diff
@@ -167,6 +172,85 @@ def evaluate_compare(input_str: str) -> Union[str, bool]:
     return input_str
 
 
+def handle_for_loop_in_foreach(input_str: str) -> str:
+    # not implementing for loop with condition for now
+    if isinstance(input_str, str) and renderer.FOR_LOOP in input_str and '?' not in input_str:
+        old_input_str = input_str
+        try:
+            e = ast.literal_eval(input_str)
```

Comment:
why are you using this evalutaion method?

Reply:
To extract some weird things from str.